File: src/MixNetFx/FacFunctions/PrintClasses/FacPrintWorkGroupListShiftCondensed.vb
Imports Facesso.Data

Public Class FacPrintWorkGroupListShiftCondensed
    Inherits FacessoPrintBase

    Private myWorkGroupAnalysis As WorkGroupAnalysisInfoItems
    Private myProductionPeriod As ProductionPeriod

    Sub New(ByVal WorkGroupAnalysis As WorkGroupAnalysisInfoItems, ByVal Period As ProductionPeriod, ByVal Username As String)
        MyBase.New("Produktiv-Site-Auswertung", WorkGroupAnalysis.Period.RangeDescription, Username)
        myWorkGroupAnalysis = WorkGroupAnalysis
        myProductionPeriod = Period
    End Sub

    Protected Overrides Sub PrepareDocument()
        MyBase.PrepareDocument()

        With PrintDocument
            .WriteLine(myProductionPeriod.ShiftParameters.ToString).DistanceToNext = 10
            'Mengentabelle der Produktiv-Site
            .CurrentFont = LayoutAndNumberFormats.TableHeaderFont.ToFont
            .BeginTable(BorderStyle, 65, 175, 135, 110, 135, 125)
            .BuildTableHeader()
            .CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopCenter
            .WriteCells("Nr.:", "Site-Name", "Brutto-" + vbNewLine + "zeiten", "Unterbr.-" + vbNewLine + "zeiten", "Netto-" + vbNewLine + "zeiten", "Kennzahlen")
            .BuildTableBody()
            .CurrentFont = LayoutAndNumberFormats.SmallTableFont.ToFont
            For Each locItem As WorkGroupAnalysisInfo In myWorkGroupAnalysis
                'Leerzeiten nicht drucken!
                If locItem.DegreeOfTime = -1 Then Continue For
                .CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopRight
                .WriteCell(locItem.WorkGroupInfo.WorkGroupNumber.ToString())
                .CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopLeft
                .WriteCell(locItem.WorkGroupInfo.WorkGroupName)
                .CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopCenter
                .WriteCell(locItem.AttendanceTimeDeltaStrings)
                .CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopRight
                .WriteCell(locItem.GeneralBreakTimeStrings)
                .CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopCenter
                .WriteCell(locItem.IncentiveTimeDeltaStrings)
                .CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopCenter
                .WriteCell("Zeitgrad: " & locItem.DegreeOfTime.ToString("##0") & vbNewLine & _
                           "Zeitgrad (angp.): " & locItem.DegreeOfTimeAdj.ToString("##0") & _
                           vbNewLine & "Auslastung:" & locItem.PercentageWorkload.ToString("#,##0.00") & " %")
            Next
            .EndTable()
        End With
    End Sub
End Class

File: src/NetFxCSharp/Facesso.Functions/PrintClasses/FacPrintWorkGroupListShiftCondensed.cs
using Facesso.Data;
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Linq;
using System.Threading.Tasks;
using System.Xml.Linq;

namespace Facesso.Functions
{
    public class FacPrintWorkGroupListShiftCondensed : FacessoPrintBase
    {
        private WorkGroupAnalysisInfoItems myWorkGroupAnalysis;
        private ProductionPeriod myProductionPeriod;
        public FacPrintWorkGroupListShiftCondensed(WorkGroupAnalysisInfoItems WorkGroupAnalysis, ProductionPeriod Period, string Username) : base("Produktiv-Site-Auswertung", WorkGroupAnalysis.Period.RangeDescription, Username)
        {
            myWorkGroupAnalysis = WorkGroupAnalysis;
            myProductionPeriod = Period;
        }

        protected override void PrepareDocument()
        {
            base.PrepareDocument();
            {
                var __with0 = PrintDocument;
                __with0.WriteLine(myProductionPeriod.ShiftParameters.ToString()).DistanceToNext = 10;
                //Mengentabelle der Produktiv-Site
                __with0.CurrentFont = LayoutAndNumberFormats.TableHeaderFont.ToFont();
                __with0.BeginTable(BorderStyle, 65, 175, 135, 110, 135, 125);
                __with0.BuildTableHeader();
                __with0.CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopCenter;
                __with0.WriteCells("Nr.:", "Site-Name", "Brutto-" + System.Environment.NewLine + "zeiten", "Unterbr.-" + System.Environment.NewLine + "zeiten", "Netto-" + System.Environment.NewLine + "zeiten", "Kennzahlen");
                __with0.BuildTableBody();
                __with0.CurrentFont = LayoutAndNumberFormats.SmallTableFont.ToFont();
                foreach (WorkGroupAnalysisInfo locItem in myWorkGroupAnalysis)
                {
                    //Leerzeiten nicht drucken!
                    if (locItem.DegreeOfTime == -1)
                    {
                        continue;
                    }

                    __with0.CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopRight;
                    __with0.WriteCell(locItem.WorkGroupInfo.WorkGroupNumber.ToString());
                    __with0.CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopLeft;
                    __with0.WriteCell(locItem.WorkGroupInfo.WorkGroupName);
                    __with0.CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopCenter;
                    __with0.WriteCell(locItem.AttendanceTimeDeltaStrings);
                    __with0.CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopRight;
                    __with0.WriteCell(locItem.GeneralBreakTimeStrings);
                    __with0.CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopCenter;
                    __with0.WriteCell(locItem.IncentiveTimeDeltaStrings);
                    __with0.CurrentCellAlignment = ActiveDev.Printing.ADTextCellAlignment.TopCenter;
                    __with0.WriteCell("Zeitgrad: " + locItem.DegreeOfTime.ToString("##0") + System.Environment.NewLine + "Zeitgrad (angp.): " + locItem.DegreeOfTimeAdj.ToString("##0") + System.Environment.NewLine + "Auslastung:" + locItem.PercentageWorkload.ToString("#,##0.00") + " %");
                }

                __with0.EndTable();
            }
        }
    }
}